contracts/contracts/strategies/sonic/SonicSwapXAMOStrategy.sol

```diff
@@ -0,0 +1,576 @@
+        (uint256 wsReservesBefore, uint256 osReservesBefore, ) = IPair(pool)
+            .getReserves();
+        // diff = wS balance - OS balance
+        int256 diffBefore = wsReservesBefore.toInt256() -
+            osReservesBefore.toInt256();
+
+        _;
+
+        // Get the asset and OToken balances in the pool
+        (uint256 wsReservesAfter, uint256 osReservesAfter, ) = IPair(pool)
+            .getReserves();
+        // diff = wS balance - OS balance
+        int256 diffAfter = wsReservesAfter.toInt256() -
+            osReservesAfter.toInt256();
+
+        if (diffBefore == 0) {
+            require(diffAfter == 0, "Position balance is worsened");
+        } else if (diffBefore < 0) {
+            // If the pool was originally imbalanced in favor of OS, then
+            // we want to check that the pool is now more balanced
+            require(diffAfter <= 0, "Assets overshot peg");
+            require(diffBefore < diffAfter, "OTokens balance worse");
+        } else if (diffBefore > 0) {
+            // If the pool was originally imbalanced in favor of wS, then
+            // we want to check that the pool is now more balanced
+            require(diffAfter >= 0, "OTokens overshot peg");
+            require(diffAfter < diffBefore, "Assets balance worse");
+        }
+    }
+
+    constructor(
+        BaseStrategyConfig memory _baseConfig,
+        address _os,
+        address _ws,
+        address _gauge
+    ) InitializableAbstractStrategy(_baseConfig) {
+        os = _os;
+        ws = _ws;
+
+        pool = _baseConfig.platformAddress;
+        gauge = _gauge;
+
+        _setGovernor(address(0));
+    }
+
+    /**
+     * Initializer for setting up strategy internal state. This overrides the
+     * InitializableAbstractStrategy initializer as SwapX strategies don't fit
+     * well within that abstraction.
+     * @param _rewardTokenAddresses Address of CRV
+     */
+    function initialize(
+        address[] calldata _rewardTokenAddresses // CRV
+    ) external onlyGovernor initializer {
+        address[] memory pTokens = new address[](1);
+        pTokens[0] = pool;
+
+        address[] memory _assets = new address[](1);
+        _assets[0] = ws;
+
+        InitializableAbstractStrategy._initialize(
+            _rewardTokenAddresses,
+            _assets,
+            pTokens
+        );
+
+        _approveBase();
+    }
+
+    /***************************************
+                    Deposit
+    ****************************************/
+
+    /**
+     * @notice Deposit Wrapped S (wS) into the SwapX pool
+     * @param _wS Address of Wrapped S (wS) contract.
+     * @param _amount Amount of Wrapped S (wS) to deposit.
+     */
+    function deposit(address _wS, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        _deposit(_wS, _amount);
+    }
+
+    function _deposit(address _wS, uint256 _wsAmount) internal {
```

Comment:
🟥 This is attackable. An attacker can imbalance the pool against us right before a deposit, with an imbalanced pool you get less LP tokens per sum(deposited tokens). After the imbalanced deposit, the attacker would arb the pool back to center, and collect the profit.